maat-court-data-api/src/main/java/gov/uk/courtdata/controller/AfterResponseClearDown.java

```diff
@@ -0,0 +1,30 @@
+package gov.uk.courtdata.controller;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.annotation.WebFilter;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@WebFilter("/*")
+@Component
+public class AfterResponseClearDown implements Filter {
+    
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+        ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        try {
+            filterChain.doFilter(servletRequest, servletResponse);
+        } finally {
+            MDC.clear();
+            log.debug("Cleared down the MDC");
```

Comment:
Filters run in an order (I'm sure you know) - I'm not exact what the default order is however we'd want to make sure that the ordering in the chain doesn't run this MDC.clear() before the filter chain is completed.

Reply:
I think by calling `MDC.clear()` after the `filterChain.doFilter(servletRequest, servletResponse)` you are essentially forcing it to run after the response has been generated. I guess we could also set the filter order to `HIGHEST_PRECEDENCE`  that way it would ensure that the `MDC.clear()` is called after the entire filter chain (request and response) has been completed?